llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -603,11 +601,7 @@ SIRegisterInfo::getMaxNumVectorRegs(const MachineFunction &MF) const {
 
     if (MinNumAGPRs == DefaultNumAGPR.first) {
       // Default to splitting half the registers if AGPRs are required.
-
-      if (MFI->mayNeedAGPRs())
-        MinNumAGPRs = MaxNumAGPRs = MaxVectorRegs / 2;
-      else
-        MinNumAGPRs = 0;
```

Comment:
I guess the removal of the forced minima yields no functional change?

Reply:
mayNeedAGPRs is a wrapper around the attribute now, this is just redundant now